inc/functions_upload.php

```diff
@@ -62,7 +62,8 @@ function remove_attachment($pid, $posthash, $aid)
 		}
 
 		$date_directory = explode('/', $attachment['attachname']);
-		if(@is_dir($uploadpath."/".$date_directory[0]))
+		$query = $db->simple_select("attachments", "COUNT(aid) as indir", "attachname LIKE '".$db->escape_string_like($date_directory[0])."/%'");
```

Comment:
Are these changes related to compatibility?

Reply:
I was getting the following warning as MyBB attempts to delete the directory every time an attachment is deleted.

![image](https://user-images.githubusercontent.com/2552068/92384724-8284e000-f108-11ea-89b9-2bbdf911f3c0.png)

It is not strictly related to compatibility with PHP 8.0 but I made the change thinking it was at the time.

I can extract into a separate PR if you prefer. I have also realised that since there is the `index.html` file in the dir then the delete directory will actually never succeed...

Reply:
I've removed the changes from this PR and opened a new PR against #4147.